File: src/main/java/utils/WeightList.java
package utils;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public final class WeightList {
    private final ArrayList<Integer> weights;
    private final IntGenerator random;

    public WeightList(@NotNull int... weights) {
        this.weights = new ArrayList<>();
        for (int weight : weights) {
            this.weights.add(weight);
        }
        random = new IntGenerator(weights.length);
    }

    public int getWeight() {
        return weights.get(random.next());
    }
}

File: src/main/java/common/WeightList.kt
package common

import java.util.ArrayList

class WeightList(vararg weights: Int) {
    private val weights = ArrayList<Int>()
    private val random: Randomer

    val weight: Int
        get() = weights[random.next()]

    init {
        for (weight in weights) {
            this.weights.add(weight)
        }
        random = Randomer(weights.size)
    }
}
